lib/elixir/lib/gen_event.ex

```diff
@@ -380,8 +380,23 @@ defmodule GenEvent do
         :gen.start(GenEvent, mode, @no_callback, [], [])
       atom when is_atom(atom) ->
         :gen.start(GenEvent, mode, {:local, atom}, @no_callback, [], [])
-      other when is_tuple(other) ->
-        :gen.start(GenEvent, mode, other, @no_callback, [], [])
+      {:global, _term} = tuple ->
+        :gen.start(GenEvent, mode, tuple, @no_callback, [], [])
+      {:via, module, _term} = tuple when is_atom(module) ->
+        :gen.start(GenEvent, mode, tuple, @no_callback, [], [])
+      other ->
+        message = """
+          expected :name option to be one of:
+
+            * nil
+            * atom
+            * {:global, term}
+            * {:via, module, term}
+
+          Got: #{inspect(other)}
+          """
+
+        raise ArgumentError, message: String.trim_trailing(message)
```

Comment:
Probably too much expressions in one line, but we can drop `message:` and it will be fine I think.